File: app/src/main/java/lt/zilinskas/marius/testio/activities/BaseActivity.java
package lt.zilinskas.marius.testio.activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

import lt.zilinskas.marius.testio.TestioApplication;

public class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    protected void initCustomToolbar(Toolbar toolbar) {
        if (toolbar == null) {
            return;
        }

        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayShowTitleEnabled(false);
    }

    public TestioApplication getTestioApplication() {
        return (TestioApplication) super.getApplication();
    }

    protected void logOut() {
        getTestioApplication().getSharedPreferences().setToken(null);
        getTestioApplication().getDatabaseHelper().getExpensesDAO().deleteAllServers();

        Intent intent = new Intent(this, LoginActivity.class);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/com/jonastiskus/testio/ui/BaseActivity.kt
package com.jonastiskus.testio.ui

import android.content.Intent
import android.os.Bundle
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.jonastiskus.testio.R
import com.jonastiskus.testio.TestioApplication
import com.jonastiskus.testio.network.service.ServiceProvider

abstract class BaseActivity : AppCompatActivity() {

    private lateinit var serviceProvider: ServiceProvider


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val testioApplication = application as TestioApplication
        serviceProvider = testioApplication.getServiceProvider()
    }

    fun enableToolbar() {
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        toolbar.findViewById<ImageView>(R.id.logout).setOnClickListener {
            val intent = Intent(this, LoginActivity::class.java)
            intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_SINGLE_TOP or Intent.FLAG_ACTIVITY_CLEAR_TASK
            startActivity(intent)
        }
    }
}
